### PR TITLE
Add IBM Cloud Pak for Data 4.0 support

### PR DESCRIPTION
**Pull request by jensmueller-com**
This pull request contains the following changes:

- Add IBM Cloud Pak for Data 4.0 support

### CODE REVIEW

File: dg/commands/cpd4/uninstall.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    """Uninstall IBM Cloud Pak for Data"""
+
+    if not force:
+        click.confirm(
+            f"Do you really want to uninstall IBM Cloud Pak for Data on server {server} from project '{project}'?",
+            abort=True,
+        )
+
+        click.confirm(
+            "Have you uninstalled any services that are installed in the same project (or tethered projects) as the "
```
Comment (johker01) [inline dg/commands/cpd4/uninstall.py:64]: Can you please elaborate what this means? I'm probably missing context, but at the moment I don't know what this prompt means and what the consequences are.

File: dg/lib/cloud_pak_for_data/cpd_4_0_0/types/cloud_pak_for_data_license.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from enum import Enum, auto
+
+
+class CloudPakForDataLicense(Enum):
```
Comment (johker01) [inline dg/lib/cloud_pak_for_data/cpd_4_0_0/types/cloud_pak_for_data_license.py:18]: Is this really the CPD license or the Db2 license? The types sound a lot like Db2 license types.

**Review by johker01 — APPROVED** [review]
I just had minor comments regarding how to use specific verbs or to add more details to prompts.

File: dg/commands/cpd4/__init__.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+import click
+
+from dg.lib.click.lazy_loading_multi_command import (
+    create_click_multi_command_class,
+)
+
+
+@click.command(cls=create_click_multi_command_class(sys.modules[__name__]))
+def cpd4():
```
[the remark this answers is not in the document]
Comment (jensmueller-com) [reply dg/commands/cpd4/__init__.py:25]: I replied to this comment on Slack.

File: dg/commands/cpd4/service/install.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    context_settings=dg.lib.click.utils.create_default_map_from_dict(
+        dg.config.cluster_credentials_manager.cluster_credentials_manager.get_current_credentials()
+    )
+)
+@click.option("--server", help="OpenShift server URL", required=True)
+@click.option("--username", help="OpenShift username")
+@click.option("--password", help="OpenShift password")
+@click.option("--token", help="OpenShift OAuth access token")
+@click.option("--installation-option", multiple=True, type=(str, CloudPakForDataServiceSpecParamType()))
+@click.option(
```
[the remark this answers is not in the document]
Comment (jensmueller-com) [reply dg/commands/cpd4/service/install.py:49]: I just followed the official [Db2 Data Gate documentation](https://www.ibm.com/docs/en/cloud-paks/cp-data/4.0?topic=iddg-installing-db2-data-gate#operator-install__install), which states that a license must be specified when creating the custom resource.

File: dg/commands/cpd4/uninstall.py (hunk 1 above)
Comment (jensmueller-com) [reply dg/commands/cpd4/uninstall.py:64]: This question corresponds to [this section](https://www.ibm.com/docs/en/cloud-paks/cp-data/4.0?topic=installing-uninstalling-cloud-pak-data#rhos-uninstall__dependencies) in the official IBM Cloud Pak for Data documentation:

> Ensure that you uninstall any services that are installed in the same project (or tethered projects) as the control plane before you uninstall the control plane.

Ideally, the CLI would know how to uninstall all instances of a service, the service itself, and the operator. To this end, we would have to statically store or dynamically determine the kind of the respective service instance (Db2 Data Gate: `DatagateServiceInstance`) and the service itself (Db2 Data Gate: `DatagateService`). According to the documentation, each operator should be uninstalled using the IBM Cloud Pak CLI (example: [Db2 Data Gate](https://www.ibm.com/docs/en/cloud-paks/cp-data/4.0?topic=gate-uninstalling-db2-data#operator-uninstall__uninstall-operator)), which is very unfortunate. Therefore, I would like to postpone this slightly complex task for now. However, feel free to suggest a better question.

File: dg/lib/cloud_pak_for_data/cpd_4_0_0/types/cloud_pak_for_data_license.py (hunk 2 above)
Comment (jensmueller-com) [reply dg/lib/cloud_pak_for_data/cpd_4_0_0/types/cloud_pak_for_data_license.py:18]: Actually, this class should be named `CloudPakForDataServiceLicense` as it contains all license names I encountered when integrating the IBM Cloud Pak for Data services.

Examples:
- **Db2**: Advanced, Community, Standard
- **Db2 Data Gate**: Enterprise, Standard

File: dg/commands/cpd4/uninstall.py (hunk 1 above)
Comment (johker01) [reply dg/commands/cpd4/uninstall.py:64]: I agree. Let's postpone this. It's definitely not necessary for a first release. Furthermore, there is always hope that the process of uninstalling a service becomes more streamlined in CPD or Openshift.

File: dg/lib/cloud_pak_for_data/cpd_4_0_0/types/cloud_pak_for_data_license.py (hunk 2 above)
Comment (johker01) [reply dg/lib/cloud_pak_for_data/cpd_4_0_0/types/cloud_pak_for_data_license.py:18]: Sounds good to me.

**Pull request merged**